src/array/ops/if_else.rs

```diff
@@ -164,3 +168,58 @@ impl PythonArray {
         todo!("[RUST-INT][PY]");
     }
 }
+
+fn from_arrow_if_then_else(
+    predicate: &BooleanArray,
+    if_true: &FixedSizeListArray,
+    if_false: &FixedSizeListArray,
+) -> DaftResult<FixedSizeListArray> {
+    let result = if_then_else(
+        predicate.downcast(),
+        if_true.downcast(),
+        if_false.downcast(),
+    )?;
+    DataArray::try_from((if_true.name(), result))
+}
+
+impl FixedSizeListArray {
+    pub fn if_else(
+        &self,
+        other: &FixedSizeListArray,
+        predicate: &BooleanArray,
+    ) -> DaftResult<FixedSizeListArray> {
+        // TODO(Clark): Support streaming broadcasting, i.e. broadcasting without inflating scalars to full array length.
+        match (predicate.len(), self.len(), other.len()) {
+            (predicate_len, self_len, other_len)
+                if predicate_len == self_len && self_len == other_len => from_arrow_if_then_else(predicate, self, other),
+            (1, self_len, 1) => from_arrow_if_then_else(
+                &predicate.broadcast(self_len)?,
+                self,
+                &other.broadcast(self_len)?,
+            ),
+            (1, 1, other_len) => from_arrow_if_then_else(
+                &predicate.broadcast(other_len)?,
+                &self.broadcast(other_len)?,
+                other,
+            ),
+            (predicate_len, 1, 1) => from_arrow_if_then_else(
+                predicate,
+                &self.broadcast(predicate_len)?,
+                &other.broadcast(predicate_len)?,
+            ),
+            (predicate_len, self_len, 1) if predicate_len == self_len => from_arrow_if_then_else(
+                predicate,
+                self,
+                &other.broadcast(predicate_len)?,
+            ),
+            (predicate_len, 1, other_len) if predicate_len == other_len => from_arrow_if_then_else(
+                predicate,
+                &self.broadcast(predicate_len)?,
+                other,
+            ),
+            (p, s, o) => {
+                Err(DaftError::ValueError(format!("Cannot run if_else against arrays with non-broadcastable lengths: self={s}, other={o}, predicate={p}")))
+            }
+        }
+    }
+}
```

Comment:
@jaychia Here is the duplicated broadcasting logic from the macro.

Reply:
Seems fine, as discussed no real need to try and force it into our current macro which really only works for the primitives...